cmd/gitserver/server/vcs_syncer_perforce.go

```diff
@@ -235,6 +292,57 @@ func p4ping(ctx context.Context, host, username, password string) error {
 	return nil
 }
 
+// p4depots returns all of the depots to which the user has access on the host
+// and whose names match the given nameFilter, which can contain asterisks (*) for wildcards
+// if nameFilter is blank, return all depots
+func p4depots(ctx context.Context, host, username, password, nameFilter string) ([]PerforceDepot, error) {
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+
+	var cmd *exec.Cmd
+	if nameFilter == "" {
+		cmd = exec.CommandContext(ctx, "p4", "-Mj", "-ztag", "depots")
+	} else {
+		cmd = exec.CommandContext(ctx, "p4", "-Mj", "-ztag", "depots", "-e", nameFilter)
+	}
+	cmd.Env = append(os.Environ(),
+		"P4PORT="+host,
+		"P4USER="+username,
+		"P4PASSWD="+password,
+	)
+
+	out, err := runWith(ctx, wrexec.Wrap(ctx, log.NoOp(), cmd), false, nil)
+	if err != nil {
+		if ctxerr := ctx.Err(); ctxerr != nil {
+			err = ctxerr
+		}
```

Comment:
Maybe this to skip the assignment and have it a little cleaner to read?

```suggestion
		if ctx.Err() != nil {
			err = ctx.Err()
		}
```

Reply:
That is cleaner looking. It might be written that way to avoid invoking `Err()` multiple times. I doubt `Err()` is an expensive function, but that's a pattern I often use when I'm not sure of the added expense of multiple invocations of a function.